run-tests.pl

```diff
@@ -170,14 +198,20 @@ sub usage
 
    -ENAME,  -ENAME=VALUE        - pass extra argument NAME or NAME=VALUE
 
-EOF
+.
 
+   write STDERR;
    exit $exitcode;
 }
 
 my $OUTPUT = first { $_->can( "FORMAT") and $_->FORMAT eq $OUTPUT_FORMAT } output_formats()
    or die "Unrecognised output format $OUTPUT_FORMAT\n";
 
+$SERVER_IMPL = 'Synapse' unless $SERVER_IMPL;
+my $hs_factory_class = first { $_->name() eq $SERVER_IMPL } homeserver_factories()
+   or die "Unrecognised server implementation $SERVER_IMPL\n";
+our $HS_FACTORY = $hs_factory_class -> new();
```

Comment:
Then down here you could simpler

    our $HS_CLASS = first { $_->NAME eq $SERVER_IMPL } homeserver_classes() or
       or die "Unrecognised server implementation $SERVER_IMPL\n";